### PR TITLE
Add version parameters for Kyma Upgrade

### PR DESCRIPTION
**Pull request by fhivemind**
**Description**

Adds version parameters and related logic for Kyma Upgrade.

### CODE REVIEW

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler_test.go [hunk 1]
```diff
@@ -61,6 +61,45 @@ func TestHandler_AttachRoutes(t *testing.T) {
 	})
 }
 
+// Testing Kyma Version is disabled due to GitHub API RATE limits
+func TestHandler_KymaVersion(t *testing.T) {
+	t.Run("kyma version validation", func(t *testing.T) {
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler_test.go:66]: You could implement a fake github client returning the response you wish to test

File: components/kyma-environment-broker/internal/process/upgrade_kyma/initialisation.go [hunk 2]
```diff
@@ -166,14 +166,27 @@ func (s *InitialisationStep) initializeUpgradeRuntimeRequest(operation internal.
...
+
+	var (
+		err     error
+		version *internal.RuntimeVersionData
+	)
+
+	// get Kyma version from orchestration or runtime parameters
+	if orchestration.HasVersion() {
+		version = internal.NewRuntimeVersionFromOrchestrationMapping(orchestration.Parameters.Version)
+	} else {
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/initialisation.go:182]: Please change this logic to 

- store supplied kyma version in each upgrade operation
-  `runtimeVerConfigurator.ForUpgrade`() uses kyma version stored in operation. Eliminate the use of orchestration during operation processing.

**Comment by PK85** [comment]
@fhivemind why do you need that feature?

**Comment by fhivemind** [comment]
This would enable more robust tests for Kyma operations and create a way for customized versioning. This PR is a dependency for a new one that will extend KCP CLI as well to allow consumption of versioning parameter.

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go [hunk 3]
```diff
@@ -101,6 +119,50 @@ func (h *kymaHandler) validateTarget(spec orchestration.TargetSpec) error {
 	return nil
 }
 
+// ValidateKymaVersion validates provided version. Supports three types of versioning:
+// semantic version, PR-<number>, and <branch name>-<commit hash>.
+// Validates version iff GitHub responded with 4xx code. If GitHub API does not work
+// (e.g. due to API RATE limit), returns version as valid.
```
Comment (polskikiel) [inline components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go:125]: Is this comment correct? From https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting I can see that after exceeding rate limit the GitHub responds with 403 so currently `shouldHandle(resp)` function will return error at 159 line.

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler_test.go [hunk 4]
```diff
@@ -61,6 +61,156 @@ func TestHandler_AttachRoutes(t *testing.T) {
...
+		require.NotNil(t, err)
+		require.Contains(t, err.Error(), "not present on branch")
+
+		err = kHandler.ValidateKymaVersion("release-0.4-f5e6d75")
+		require.NotNil(t, err)
+		require.Contains(t, err.Error(), "not present on branch")
+	})
+}
+
+func fixKymaHanler(t *testing.T) *kymaHandler {
```
Comment (polskikiel) [inline components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler_test.go:100]: typo

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go [hunk 5]
```diff
@@ -50,13 +58,23 @@ func (h *kymaHandler) createOrchestration(w http.ResponseWriter, r *http.Request
...
+	// validate target
 	err := h.validateTarget(params.Targets)
 	if err != nil {
 		h.log.Errorf("while validating target: %v", err)
 		httputil.WriteErrorResponse(w, http.StatusBadRequest, errors.Wrapf(err, "while validating target"))
 		return
 	}
 
+	// validate Kyma version
+	err = h.ValidateKymaVersion(params.Version)
```
Comment (polskikiel) [inline components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go:71]: can we just omit this check if version wasn't provided?

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go (hunk 3 above)
Comment (fhivemind) [reply components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go:125]: Good observation @polskikiel, I will patch it up in a bit. Thanks!

File: components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go (hunk 5 above)
Comment (fhivemind) [reply components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go:71]: In my opinion, not quite. We should have log details if possible when the validation fails.

**Comment by PK85** [comment]
@fhivemind please provide examples from real world about "more robust tests for Kyma operations". We do not want to have customized versioning on production. All SKRs must have same newest version. Please ping me on slack, and we can discuss that PR.

**Review by polskikiel — APPROVED** [review]

**Pull request merged**